cw-rpm.mk

```diff
@@ -0,0 +1,187 @@
+# REPO_SERVER          - username and server to scp pacakges to (default is unset,
+#                        meaning move packages locally)
+# CW_SIGNED            - whether to sign the generated package repository (default
+#                        is unset, meaning don't sign; set to Y to sign).
+#                        IMPORTANT: this signs the repo itself, which means that
+#                        *all* packages in it are marked authentic, not just the
+#                        ones we built just now.
+# HARDENED_REPO_DIR    - path to the hardened repository to move packages to
+#                        (default is unset, meaning don't move packages)
+# HARDENED_REPO_SERVER - username and server for the hardened repository to scp
+#                        packages to (default is unset, meaning move
+#                        packages locally)
+
+# Include common definitions
+include build-infra/cw-pkg.mk
+
+# Default RPM_* from PKG_*
+RPM_COMPONENT ?= $(PKG_COMPONENT)
+RPM_MAJOR_VERSION ?= $(PKG_MAJOR_VERSION)
+RPM_MINOR_VERSION ?= $(PKG_MINOR_VERSION)
+RPM_NAMES ?= $(PKG_NAMES)
+
+RPM_ARCH := $(shell rpmbuild -E %{_arch})
+
+# Commands to build an RPM package repo.
+RPM_BUILD_REPO := createrepo .
+ifeq ($(CW_SIGNED), Y)
+RPM_BUILD_REPO := $(RPM_BUILD_REPO) ; gpg -abs -u $(CW_SIGNER) repodata/repomd.xml
+endif
+
+# Build and move to the repository server (if present).
+.PHONY: rpm-only
+rpm-only: rpm-build rpm-move rpm-move-hardened
+
+# Build the .rpm files in rpm/RPMS.
+.PHONY: rpm-build
+rpm-build:
+	# Clear out old RPMs
+	rm -rf rpm/SRPMS/*
+	rm -rf rpm/RPMS/$${RPM_ARCH}/*
+	# If this is built from a git@github.com: URL then output Git instructions for accessing the build tree
+	echo "* $$(date -u +"%a %b %d %Y") $(CW_SIGNER_REAL) <$(CW_SIGNER)>" >rpm/changelog;\
+	if [[ "$$(git config --get remote.origin.url)" =~ ^git@github.com: ]]; then\
+	        echo "- built from $$(git config --get remote.origin.url|sed -e 's#^git@\([^:]*\):\([^/]*\)\([^.]*\)[.]git#https://\1/\2\3/tree/#')$$(git rev-parse HEAD)" >>rpm/changelog;\
+		echo "  Use Git to access the source code for this build as follows:" >>rpm/changelog;\
+		echo "    $$ git config --global url.\"https://github.com/\".insteadOf git@github.com:" >>rpm/changelog;\
+		echo "    $$ git clone --recursive $$(git config --get remote.origin.url)" >>rpm/changelog;\
+		echo "    Cloning into '$$(git config --get remote.origin.url|sed -e 's#^\([^:]*\):\([^/]*\)/\([^.]*\)[.]git#\3#')'..." >>rpm/changelog;\
+		echo "      ..."  >>rpm/changelog;\
+		echo "    $$ cd $$(git config --get remote.origin.url|sed -e 's#^\([^:]*\):\([^/]*\)/\([^.]*\)[.]git#\3#')" >>rpm/changelog;\
+		echo "    $$ git checkout -q $$(git rev-parse HEAD)" >>rpm/changelog;\
+		echo "    $$ git submodule update --init" >>rpm/changelog;\
+		echo "      ..."  >>rpm/changelog;\
+		echo "    $$"  >>rpm/changelog;\
+        else\
+		echo "- built from revision $$(git rev-parse HEAD)" >>rpm/changelog;\
+	fi
```

Comment:
It's similar, but not quite the same - changelog file format differs between Debian and RPM (in terms of bullet and whitespace style).